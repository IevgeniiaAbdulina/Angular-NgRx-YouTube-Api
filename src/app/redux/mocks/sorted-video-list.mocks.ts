import { VideoItem } from 'src/app/shared/models/video-list-response';

export const sortedVideoListByDateMock: VideoItem[] = [
  {
    kind: 'youtube#searchResult',
    etag: 'j7Nz5_A6m-rRRF3wv35-gfA80rk',
    id: 'CSU7SggjVwg',
    snippet: {
      publishedAt: '2023-12-04T20:45:41Z',
      channelId: 'UCPgzHeVeVDsUvcvovWHjHGQ',
      title:
        'Różaniec Teobańkologia z prorokiem Izajaszem o górach i przepaściach 4.12 Poniedziałek',
      description:
        'Różaniec Teobańkologia odmawiamy codziennie na żywo o 20:30. Dziś rozważamy tajemnice radosne. Rozważania różańcowe ...',
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/CSU7SggjVwg/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '1',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  },
  {
    kind: 'youtube#searchResult',
    etag: '5788M5Le5PZfiXxLQkkKfLxxJYo',
    id: '_iyVsT2s2xg',
    snippet: {
      publishedAt: '2023-12-04T20:37:00Z',
      channelId: 'UCDfk8ogO6QGeJAYCY0QDzKw',
      title: 'My TESA Autopilot Failed!!',
      description:
        'My brand new Tesla auto pilot failed. Just like the cyber truck. My Tesla is brand brand new and it has auto pilot full self driving.',
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/_iyVsT2s2xg/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '1',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  }
];

export const videoListSortedByViewsMock: VideoItem[] = [
  {
    kind: 'youtube#searchResult',
    etag: 'RfFF-gdDOPOms7GUgui1Rdf2mFY',
    id: 'YbQVrJXfuJM',
    snippet: {
      publishedAt: '2018-05-24T12:29:37Z',
      channelId: 'UC5ljgk-ZnRBrC4fnbSGVsDQ',
      title: 'BLUE BIRD | cały film | PL | short love movie',
      description:
        'Film powstały we współpracy z Warszawską Szkołą Filmową i Polskim Instytutem Sztuki Filmowej. KONTAKT: ...',
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/YbQVrJXfuJM/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '100',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  },
  {
    kind: 'youtube#searchResult',
    etag: '3CKhG34WA4J9DiWR5fkxGBJ3Fe8',
    id: 'N-PyWfVkjZc',
    snippet: {
      publishedAt: '2020-11-26T17:00:11Z',
      channelId: 'UCsfdrdj6DAWmBUdPDfFTatg',
      title:
        'Shakin&#39; Stevens - Merry Christmas Everyone (Official 4K Video)',
      description:
        "Shakin' Stevens - Merry Christmas Everyone | Stream Shakin' Stevens here: https://shakinstevens.lnk.to/MCE2022ID Listen to all ...",
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/N-PyWfVkjZc/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '50',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  }
];
