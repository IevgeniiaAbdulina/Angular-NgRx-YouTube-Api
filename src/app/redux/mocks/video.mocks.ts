import { CustomVideo } from 'src/app/shared/models/custom-video-interface';
import { VideoItem } from 'src/app/shared/models/video-list-response';

import { CustomVideoState, VideoState } from '../state.models';

export const videoItemMock: VideoItem = {
  id: '123',
  snippet: {
    publishedAt: '12/2/2023',
    channelId: '',
    title: 'First mock video item',
    description: 'Test video item',
    thumbnails: {
      high: {
        url: 'assets/img/default_image.svg',
        width: 100,
        height: 150
      }
    },
    channelTitle: 'testing',
    tags: ['1'],
    categoryId: '',
    liveBroadcastContent: '',
    defaultLanguage: '',
    localized: {
      title: '',
      description: ''
    },
    defaultAudioLanguage: ''
  },
  statistics: {
    viewCount: '1',
    likeCount: '2',
    dislikeCount: '3',
    favoriteCount: '4',
    commentCount: '5'
  },
  kind: 'mockYoutube#video',
  etag: '1'
};

export const videoListMock: VideoItem[] = [videoItemMock, videoItemMock];

export const favoriteListMock: VideoItem[] = [
  videoItemMock,
  videoItemMock,
  videoItemMock
];

export const customVideoItemMock: CustomVideo = {
  id: '456',
  snippet: {
    publishedAt: '12/2/2023',
    channelId: '',
    title: 'First mock video item',
    description: 'Test video item',
    thumbnails: {
      high: {
        url: 'assets/img/default_image.svg',
        width: 100,
        height: 150
      }
    },
    channelTitle: 'testing',
    tags: ['1'],
    categoryId: '',
    liveBroadcastContent: '',
    defaultLanguage: '',
    localized: {
      title: '',
      description: ''
    },
    defaultAudioLanguage: ''
  },
  statistics: {
    viewCount: '1',
    likeCount: '2',
    dislikeCount: '3',
    favoriteCount: '4',
    commentCount: '5'
  },
  kind: 'mockCustom#video',
  etag: '1'
};

export const customVideoListMock: CustomVideo[] = [
  customVideoItemMock,
  customVideoItemMock
];

export const videoListStateMock: VideoState = {
  isLoading: false,
  videoList: videoListMock,
  favoriteList: videoListMock,
  error: ''
};

export const customVideoStateMock: CustomVideoState = {
  isLoading: false,
  customVideoList: customVideoListMock,
  error: ''
};
