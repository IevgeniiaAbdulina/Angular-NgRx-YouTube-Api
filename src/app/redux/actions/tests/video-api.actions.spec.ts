import { videoListMock } from '../../mocks/video.mocks';
import {
  getVideoList,
  getVideoListSuccess,
  setLoading
} from '../video-api.actions';

describe('VideoActions', () => {
  describe('SetLoading', () => {
    it('should set loading state', () => {
      const action = setLoading({ isLoading: true });

      expect(action.isLoading).toEqual(true);
      expect(action).toMatchSnapshot();
    });
  });

  describe('GetVideoList', () => {
    it('should create an action to get video list', () => {
      const expectedAction = {
        type: getVideoList.type
      };
      const action = getVideoList();

      expect(action).toEqual(expectedAction);
    });
  });

  describe('GetVideoListSuccess', () => {
    it('should create an action to get video list success', () => {
      const expectedAction = {
        type: getVideoListSuccess.type,
        videoList: videoListMock
      };
      const action = getVideoListSuccess({
        videoList: videoListMock
      });

      expect(action).toEqual(expectedAction);
    });
  });
});
