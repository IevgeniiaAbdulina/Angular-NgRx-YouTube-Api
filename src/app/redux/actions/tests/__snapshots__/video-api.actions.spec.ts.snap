// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoActions SetLoading should set loading state 1`] = `
{
  "isLoading": true,
  "type": "[Video List] Get Loading State",
}
`;
